src/commands/transaction/transfer.ts

```diff
@@ -108,42 +120,82 @@ export default class extends AnnounceTransactionsCommand {
             'message',
             () => '',
             'Introduce the message: ');
-
-        let message: PlainMessage | EncryptedMessage;
-
         if (options.message) {
             options.encrypted = options.encrypted ? options.encrypted : readlineSync.keyInYN(
                 'Do you want to send an encrypted message?');
         }
-
-        if (options.message && options.encrypted) {
-            options.recipientPublicKey = OptionsResolver(options,
-                'recipientPublicKey',
-                () => undefined,
-                'Introduce the recipient public key: ');
-            message = profile.account.encryptMessage(
-                options.message,
-                PublicAccount.createFromPublicKey(options.recipientPublicKey,
-                    profile.networkType),
-                profile.networkType);
-        } else {
-            message = PlainMessage.create(options.message);
-        }
-
         options.maxFee = OptionsResolver(options,
             'maxFee',
             () => undefined,
             'Introduce the maximum fee you want to spend to announce the transaction: ');
 
-        const transferTransaction = TransferTransaction.create(
-            Deadline.create(),
-            recipient,
-            mosaics,
-            message,
-            profile.networkType,
-            options.maxFee ? UInt64.fromNumericString(options.maxFee) : UInt64.fromUint(0));
+        if (true !== options.persistentHarvestingDelegation) {
+            let message: PlainMessage | EncryptedMessage;
+            if (options.message && options.encrypted) {
+                options.recipientPublicKey = OptionsResolver(options,
+                    'recipientPublicKey',
+                    () => undefined,
+                    'Introduce the recipient public key: ');
+                message = profile.account.encryptMessage(
+                    options.message,
+                    PublicAccount.createFromPublicKey(options.recipientPublicKey,
+                        profile.networkType),
+                    profile.networkType);
+            } else {
+                message = PlainMessage.create(options.message);
+            }
+
+            const transferTransaction = TransferTransaction.create(
+                Deadline.create(),
+                recipient,
+                mosaics,
+                message,
+                profile.networkType,
+                options.maxFee ? UInt64.fromNumericString(options.maxFee) : UInt64.fromUint(0));
+
+            const signedTransaction = profile.account.sign(transferTransaction, profile.networkGenerationHash);
+            this.announceTransaction(signedTransaction, profile.url);
+        } else {
```

Comment:
It is not necessary to recreate the whole transaction; you can send a TransferTransaction using a ``PersistentHarvestingDelegationMessage`` as we are doing for the encrypted message.